theano/gpuarray/dnn.py

```diff
@@ -2833,6 +2833,295 @@ def local_abstractconv3d_cudnn_graph(op, context_name, inputs, outputs):
     return [rval]
 
 
+class GpuDnnTransformerDescriptor(COp):
+
+    """
+    This Op builds a spatial transformer descriptor for use in spatial transformer network
+    operations.
+    """
+
+    __props__ = ('dtype',)
+    params_type = ParamsType(dtype=cudnn.cudnnDataType_t)
+
+    def c_headers(self):
+        return ['cudnn.h', 'cudnn_helper.h']
+
+    def c_header_dirs(self):
+        header_dirs = [os.path.dirname(__file__)]
+        if config.dnn.include_path:
+            header_dirs += [config.dnn.include_path]
+        return header_dirs
+
+    def c_libraries(self):
+        return ['cudnn']
+
+    def c_lib_dirs(self):
+        lib_dirs = []
+        if config.dnn.library_path:
+            lib_dirs += [config.dnn.library_path]
+        return lib_dirs
+
+    def do_constant_folding(self, node):
+        return False
+
+    def __init__(self, dtype=theano.config.floatX):
+        COp.__init__(self, ["c_code/dnn_sptf_desc.c"], "APPLY_SPECIFIC(dnn_sptf_desc)")
+        assert cudnn.cudnnDataType_t.has_alias(dtype)
+        self.dtype = dtype
+
+    def make_node(self, dimensions):
+        dimensions = as_tensor_variable(dimensions)
```

Comment:
You'are absolutely correct, I'll follow your suggestions, of course. I've used int types, because that is how the interface of cuDNN is specified, but it is essential to check the Python object, before reading from it in the C code.